File: Python/Manual 03/Question03.py
def print_diamond(n):
    i = 1
    while i <= n:
        k = 1
        while k <= n - i:
            print(" ", end="")
            k += 1

        j = 1
        while j <= i:
            print("* ", end="")
            j += 1
        print()
        i += 1

    i = 1
    while i < n:
        j = n - 1
        while j >= n - i:
            print(" ", end="")
            j -= 1

        k = 1
        while k <= n - i:
            print("* ", end="")
            k += 1
        print()
        i += 1


n = int(input("Enter number of rows: "))
print_diamond(n)

File: C++/Manual 03/Question03.cpp
#include <iostream>
using namespace std;

int main()
{
    // Diamond
    int i, j, k, n;
    cout << "Enter number of rows: ";
    cin >> n;
    i = 1;
    while (i <= n)
    {
        k = 1;
        while (k <= n - i)
        {
            cout << " ";
            k++;
        }
        j = 1;
        while (j <= i)
        {
            cout << "* ";
            j++;
        }
        cout << endl;
        i++;
    }
    i = 1;
    while (i < n)
    {
        j = n - 1;
        while (j >= n - i)
        {
            cout << " ";
            j--;
        }
        k = 1;
        while (k <= n - i)
        {
            cout << "* ";
            k++;
        }
        cout << endl;
        i++;
    }
}
